app/tests/test_enterprise.py

```diff
@@ -93,3 +94,29 @@ def test_grant_permissions(client: TestClient):
         cookies=client.cookies.get_dict(),
     )
     assert r_enterprise_grant_access.status_code == HTTP_204_NO_CONTENT
+
+def test_grant_permissions_unauthorised(client: TestClient):
+    user_header = get_random_logged_user(client)
+    r_enterprise = create_enterprise(client)
+    r_enterprise_grant_access = client.post(
+        f"/enterprise/{r_enterprise.json()['id']}/access",
+        json={"user_id": 1, "role_to_grant": "VIEWER"},
+    )
+    assert r_enterprise_grant_access.status_code == HTTP_401_UNAUTHORIZED
+
+def test_grant_permissions_forbidden(client: TestClient):
+    user_header = get_random_logged_user(client)
+    r_enterprise = create_enterprise(client)
+    user2_register = register_sample_user(client, email="sample@email.com", password="somepass")
+    r1 = client.post(
+        f"/enterprise/{r_enterprise.json()['id']}/access",
+        json={"user_id": 2, "role_to_grant": "EDITOR"},
+        cookies=client.cookies.get_dict(),
+    )
+    user2_login = login_sample_user(client, "sample@email.com", "somepass")
+    r2 = client.post(
+        f"/enterprise/{r_enterprise.json()['id']}/access",
+        json={"user_id": 1, "role_to_grant": "VIEWER"},
+        cookies=client.cookies.get_dict(),
+    )
+    assert r2.status_code == HTTP_403_FORBIDDEN
```

Comment:
@eredzik here I get http 200 instead of 403

Reply:
It needs else statement on endpoint - see comment